### PR TITLE
Change: started to baseset generation in CMake

### PR DESCRIPTION
**Pull request by glx22**
The easy part seems to be working :)

### CODE REVIEW

**Comment by glx22** [comment]
Hmm it doesn't work if a baseset file is removed or added

**Comment by glx22** [comment]
My code is currently using list(JOIN) command but it's only available since version 3.12, I guess I'll need to change that part if we're aiming for 3.5.

I also used a workaround for nasty list expansions in add_custom_command(), while a clean solution (COMMAND_EXPAND_LISTS) is available since 3.8.

File: media/baseset/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_win.obg
+        ${CMAKE_CURRENT_SOURCE_DIR}/no_music.obm
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_dos.obm
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_tto.obm
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_win.obm
+        ${CMAKE_CURRENT_SOURCE_DIR}/no_sound.obs
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_dos.obs
+        ${CMAKE_CURRENT_SOURCE_DIR}/orig_win.obs
+)
+set(BASESET_EXTRAGRF_FILE ${CMAKE_SOURCE_DIR}/bin/baseset/orig_extra.grf)
```
Comment (TrueBrain) [inline media/baseset/CMakeLists.txt]: please avoid temporary variables in CMake. They often lead to more confusion than they resolve.

**Comment by TrueBrain** [comment]
Yeah, we really should be 3.5 .. lists often can be fixed with replacing ' ' to ';' and the other way around. Is that enough to remove 3.6+ dependencies?

**Comment by glx22** [comment]
I think all the code is now 3.5 only

**Comment by TrueBrain** [comment]
Added this to the branch. We can fix things from there.

**Pull request closed without merging**